R/github-pat.R

```diff
@@ -66,6 +69,36 @@ set_github_pat <- function(force_new = FALSE, validate = interactive(), verbose
   return(FALSE)
 }
 
+# Some notes:
+# - Format: GITHUB_PAT_({user}_AT_){host}
+# - By default, tokens get stored as user 'PersonalAccessToken' in the credential
+#   store, to distinguish them from user/password credentials that the user may have
+#   stored. You can override this via the URL, but better not.
+# - GitHub usernames (incl GHE) can only contain alphanumeric characters and dashes.
+# - Hostnames are case insensitive. We normalize hostname tolower, so you can
+#   safely parse _AT_ to look for custom usernames.
+get_pat_credential_url <- function(url){
+  # strip protocol
+  url <- sub("^[a-z]+://", "", tolower(url))
```

Comment:
This `tolower()` is in tension with what gh has been doing when forming these env var names.